### PR TITLE
#149 Be able to evaluate page with iframes

### PR DESCRIPTION
**Pull request by davidhsv**
Disabling webSecurity in phantomJs -> so we can get the document root from frames.

### CODE REVIEW

**Comment by davidhsv** [comment]
I signed it!

File: test/js/audit-rule-test.js [hunk 1]
```diff
@@ -41,6 +41,16 @@
         equal(matched.length, DIV_COUNT);
     });
 
+    test("Iframe with simple DOM", function () {
+        var ifrm = document.createElement("IFRAME");
+        var container = document.getElementById('qunit-fixture');
+        container.appendChild(ifrm);
+        ifrm.contentDocument.body.appendChild(buildTestDom());
+        var matched = [];
+        axs.AuditRule.collectMatchingElements(container, matcher, matched);
```
Comment (ckundo) [inline test/js/audit-rule-test.js]: Is this method mutating `matched` in memory? That doesn't seem ideal. Unrelated to this PR, but since we're in the neighborhood I thought I'd point it out.

File: src/audits/FocusableElementNotVisibleAndNotAriaHidden.js [hunk 2]
```diff
@@ -42,7 +42,8 @@ axs.AuditRules.addRule({
         }
         // Ignore elements which have a negative tabindex and no text content,
         // as they will be skipped by assistive technology
-        if (axs.properties.findTextAlternatives(element, {}).trim() === '')
+        var alternative = axs.properties.findTextAlternatives(element, {});
```
Comment (ckundo) [inline src/audits/FocusableElementNotVisibleAndNotAriaHidden.js]: This doesn't seem related to this PR.

File: src/js/AuditRule.js [hunk 3]
```diff
@@ -182,6 +182,14 @@ axs.AuditRule.collectMatchingElements = function(node, matcher, collection,
             }
         }
     }
+
+    //If it is a iframe, get the contentDocument
+    if (element && element.localName == 'iframe' && element.contentDocument) {
+        axs.AuditRule.collectMatchingElements(element.contentDocument,
+                                              matcher,
+                                              collection,
+                                              opt_shadowRoot);
```
Comment (ckundo) [inline src/js/AuditRule.js]: Is `opt_shadowRoot` necessary here? Is it required for `collectMatchingElements`?

**Comment by ckundo** [comment]
Check out my comment on the `opt_shadowRoot`. Aside from that question, LGTM.

**Comment by ckundo** [comment]
Actually, forgot to also mention that I don't think we need to rebuild `axs_testing.js` for this to work. That change should be reverted I think.

File: src/audits/FocusableElementNotVisibleAndNotAriaHidden.js (hunk 2 above)
Comment (davidhsv) [inline src/audits/FocusableElementNotVisibleAndNotAriaHidden.js]: Just a cleanup, I was getting a exception sometimes when findTextAlternatives return null (null.trim()).

File: src/js/AuditRule.js (hunk 3 above)
Comment (davidhsv) [inline src/js/AuditRule.js]: I think so, it's a recursive method and I think it's necessary to always pass this atribute forward. For example, if we got a shadow down inside a iframe contentDocument, it will be needed.

File: test/js/audit-rule-test.js (hunk 1 above)
Comment (davidhsv) [inline test/js/audit-rule-test.js]: I was justing copying the same behavior in others tests (in this same file). I don't like methods changing parameter's content too (bad smell).

**Comment by davidhsv** [comment]
The axs_testing.js was generated by the grunt build. Should I revert it?

**Comment by davidhsv** [comment]
Reverted file ax_testing.js to match remote master.

File: dist/js/axs_testing.js [hunk 4]
```diff
@@ -1545,7 +1545,7 @@ axs.AuditRule.collectMatchingElements = function(a, b, c, d) {
         for (e = f.getDistributedNodes(), f = 0;f < e.length;f++) {
```
Comment (ckundo) [inline dist/js/axs_testing.js]: @davidhsv this file needs to be checked out entirely.

File: src/js/AuditRule.js (hunk 3 above)
Comment (ckundo) [inline src/js/AuditRule.js]: Makes sense.

File: test/js/audit-rule-test.js (hunk 1 above)
Comment (ckundo) [inline test/js/audit-rule-test.js]: Yep, that's fine, thought I'd point it out as something we can improve in the future.

**Comment by ckundo** [comment]
@davidhsv looks like there's a merge conflict. Once that's resolved and you remove `axs_testing.js` changes, LGTM.

**Comment by alice** [comment]
I've fixed the Gruntfile so that it won't create spurious changes to `dist/axs_testing.js` any more.

File: src/audits/FocusableElementNotVisibleAndNotAriaHidden.js (hunk 2 above)
Comment (alice) [inline src/audits/FocusableElementNotVisibleAndNotAriaHidden.js]: Good catch! I just submitted a separate PR for this same issue, not having seen this, sorry.

File: test/js/audit-rule-test.js (hunk 1 above)
Comment (alice) [inline test/js/audit-rule-test.js]: This is intended behaviour. Since this is a recursive method, we need an accumulator to keep track of matched elements - otherwise we'd be constantly having to combine partial results.

**Comment by alice** [comment]
Apologies for coming late to this. This LGTM :+1: once the merge issues are fixed.

**Comment by ckundo** [comment]
@davidhsv bump. Can you resolve the merge conflicts and push again? Thanks!

**Comment by davidhsv** [comment]
@ckundo Done :+1:
